ValgrindCI/render.py

```diff
@@ -121,13 +123,17 @@ def _extract_error_data(
                 assert error_line is not None
                 stack["line"] = error_line - lines_before - 1
                 stack["error_line"] = lines_before + 1
-                stack["fileref"] = "{}:{}".format(
-                    frame.get_path(self._source_dir), error_line
-                )
-                with open(fullname, "r") as f:
-                    for l, code_line in enumerate(f.readlines()):
-                        if l >= stack["line"] and l <= error_line + lines_after - 1:
-                            stack["code"].append(code_line)
+                frame_source = frame.get_path(self._source_dir)
+                frame_source = self._data.relativize(frame_source)
+                stack["fileref"] = "{}:{}".format(frame_source, error_line)
+                fullname = self._data.substitute_path(fullname)
+                try:
+                    with open(fullname, "r", errors="replace") as f:
+                        for l, code_line in enumerate(f.readlines()):
+                            if l >= stack["line"] and l <= error_line + lines_after - 1:
+                                stack["code"].append(code_line)
+                except OSError as e:
+                    print(f"Warning: cannot read stack data from missing source file: {e.filename}")
```

Comment:
Shouldn't `continue` be called when swallowing this exception in order to avoid adding a malformed "stack" to the `issue` dictionary ?
```suggestion
                    print(f"Warning: cannot read stack data from missing source file: {e.filename}")
                    continue
```